### PR TITLE
Adds SHOULD_NOT_SLEEP to Destroy. Why didn't we do this before.

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

Mostly sane INVOKE_ASYNC stuff, ~~outside of one UNLINT that I can't get around~~ mothblocks wuz here -Fuck you mothman

## Why It's Good For The Game

image

### CODE REVIEW

File: code/modules/modular_computers/computers/item/computer.dm [hunk 1]
```diff
@@ -417,7 +417,7 @@
 		active_program = null
 	var/mob/user = usr
 	if(user && istype(user))
-		ui_interact(user) // Re-open the UI on this computer. It should show the main screen now.
+		UNLINT(INVOKE_ASYNC(src, /obj/item/modular_computer/ui_interact, user)) // Re-open the UI on this computer. It should show the main screen now.
```
Comment (Mothblocks) [inline code/modules/modular_computers/computers/item/computer.dm:420]: What couldn't you figure out? Should at least be commented.

Comment (Mothblocks) [reply code/modules/modular_computers/computers/item/computer.dm:420]: Like, I'm pretty sure you can just `/datum/proc/ui_interact` and you're fine.

Comment (LemonInTheDark) [reply code/modules/modular_computers/computers/item/computer.dm:420]: byond doesn't like .proc/ui_interact, and it doesn't like /obj/item/modular_computer/proc/ui_interact, but smdmm doesn't like a proc path that goes nowhere

Comment (LemonInTheDark) [reply code/modules/modular_computers/computers/item/computer.dm:421]: well uh

File: code/game/objects/structures/kitchen_spike.dm [hunk 2]
```diff
@@ -128,13 +128,13 @@
...
 	unbuckle_mob(M,force=1)
-	M.emote("scream")
+	INVOKE_ASYNC(M, /mob/proc/emote, "scream")
 	M.AdjustParalyzed(20)
 
 /obj/structure/kitchenspike/Destroy()
 	if(has_buckled_mobs())
-		for(var/mob/living/L in buckled_mobs)
-			release_mob(L)
+		for(var/mob/living/spiked_lad in buckled_mobs)
```
Comment (Ghilker) [inline code/game/objects/structures/kitchen_spike.dm:136]: Look at you, cleaning up!

**Review by Cyberboss — COMMENTED** [review]
Just checking, none of the `INVOKE_ASYNC`'d procs will fail due to the caller being deleted right? `malfvacate` in particular sounds sus to me.

**Comment by LemonInTheDark** [comment]
> 
> 
> Just checking, none of the `INVOKE_ASYNC`'d procs will fail due to the caller being deleted right? `malfvacate` in particular sounds sus to me.

They would, but aren't invoke_async'd procs instantly called? they're async in the sense that they block sleeps from propagating up the chain, but I thought that was the limit of their effect

**Comment by Cyberboss** [comment]
True, I meant after the sleeps

**Comment by LemonInTheDark** [comment]
damn it I didn't consider that. Wow I'm stupid.

I'll do a pass then yeah. Guess the main issue is sleeping at all in destroy is asking to hang refs. I'll see about making sure they're all cleared away before it's done

**Review by LemonInTheDark — COMMENTED** [review]
Some explanation on the two things I'm having issues with. One seems semi simple to solve, may bully kyler about the other, not sure yet

**Comment by LemonInTheDark** [comment]
Kyler tells me it's alright

**Pull request merged**